Update RFC process (#2716)
model-checking/kani: rfc/src/template.md

> - **Feature Name:** *Fill me with pretty name and a unique ident [^unstable_feature] (E.g: New Feature (`new_feature`)*

Reviewer: ```suggestion
- **Feature Name:** *Fill me with pretty name and a unique ident [^unstable_feature]. Example: New Feature (`new_feature`)*
```
Noted that the parenthesis doesn't close.

## Summary

Short (1-2 sentences) description of the feature. What is this feature about?

## User Impact

Why are we doing this? How will this benefit the final user?

 - If this is an API change, how will that impact current users?
 - If this RFC is related to change in the architecture without major user impact, think about the long term

> impact for user. I.e.: what future work will this enable. Maybe reach out to other developers to ensure you  need an RFC in the first place.

Reviewer 2: Double space.

And I would say "If you are unsure, reach out to the Kani developers ..." and perhaps even clarify that that "reaching out" should probably be a feature request issue.

## User Experience

This should be a description on how users will interact with the feature.

> Users should be able to read this section and understand how to use the feature.

Author: I prefer avoiding links to old documents as mentioned before. We have a template so it reflects our current preference based on what we are learning from previous RFCs.

That said, if others feel strong about adding examples, I do prefer adding them here since we can reduce the scope of the example, and keep it coherent with the template.

**Do not include implementation details in this section, neither discuss the rationale behind the chosen UX.**

Please include:
  - High level user flow description
  - Any new major functions or attributes that will be added to Kani library.
  - New command line options or subcommands (no need to mention the unstable flag).

>   - What are possible user errors and how are they presented. E.g: Compilation vs Verification error.

Author: ```suggestion
  - List failure scenarios and how are they presented (e.g., compilation errors, verification failures, and possible failed user iterations).
```

  - Substantial changes to existing functionality or Kani output.

If the RFC is related to architectural changes and there are no visible changes to UX, please state so.
No further explanation is needed.

## Software Design

> **We recommend you to leave this empty for the first version of your RFC**.

Reviewer 2: I would not lead with this. I would lead with the one-sentence-summary, e.g. "This is the technical ...., then say that this should only contain any major changes to the architecture and then add the "We recommend you to leave this empty ..." sentence

Author: Sure. That's what I had initially. :)

This is the technical portion of the RFC. Please provide a high level description your design.

- What are the main components that will be modified? (E.g.: changes to `kani-compiler`, `kani-driver`, metadata, proc-macros, installation...)
- Will there be changes to the components interface?
- Any changes to how these components communicate?
- What corner cases do you anticipate?

## Rationale and alternatives

This is the section where you discuss the decisions you made.

- What are the pros and cons of the UX? What would be the alternatives?
- What is the impact of not doing this?
- Any pros / cons on how you designed this?

## Open questions

List of open questions + an optional link to an issue that capture the work required to address the open question. Capture the details of the open question in the issue, not here.

Example:
- Is there any use case that isn't handled yet?

> - Is there any part of the UX that still need some improvement?

Reviewer 3: ```suggestion
- Is there any part of the UX that still needs some improvement?
```

Make sure all open questions are addressed before stabilization.

## Out of scope / Future Improvements

List of extensions and possible improvements that you predict for this feature that is out of the
scope of this RFC.

> Feel free to add as many items as you want, but please refrain from adding too much detail. If you want to capture your thoughts or start a discussion, please create a feature request. You are welcome to add a link to the new issue here.

Reviewer 2: I would maybe even add that this section is optional. Especially someone submitting a first PR should not focus on what comes later.

## Summary

Short (1-2 sentences) description of the feature. What is this feature about?

## User Impact

Imagine this as your elevator pitch directed to users as well as Kani developers.

> Why are we doing this?
> Why should users care about this feature?
> How will this benefit them?
> What is the downside?

Reviewer: ```suggestion
- Why are we doing this?
- Why should users care about this feature?
- How will this benefit them?
- What is the downside?
```

  - Substantial changes to existing functionality or Kani output.

If the RFC is related to architectural changes and there are no visible changes to UX, please state so.
No further explanation is needed.

## Software Design

This is the beginning of the technical portion of the RFC.
From now on, your main audience is Kani developers, so it's OK to assume readers know Kani architecture.

Please provide a high level description your design.

- What are the main components that will be modified? (E.g.: changes to `kani-compiler`, `kani-driver`, metadata, proc-macros, installation...)
- Will there be changes to the components interface?
- Any changes to how these components communicate?
- What corner cases do you anticipate?

**We recommend you to leave this empty for the first version of your RFC**.

## Rationale and alternatives

This is the section where you discuss the decisions you made.

- What are the pros and cons of the UX? What would be the alternatives?
- What is the impact of not doing this?
- Any pros / cons on how you designed this?

## Open questions

> List of open questions + an optional link to an issue that capture the work required to address the open question.

Reviewer: ```suggestion
List of open questions + an optional link to an issue that captures the work required to address the open question.
```
